java/sage/MediaServerRemuxer.java

```diff
@@ -884,9 +887,9 @@ private void pushData(byte[] buffer, int offset, int length)
       if (buffer[offset] != 0x47 || !tsSynced)
       {
         int startingOffset = offset;
-        if (Sage.DBG) System.out.println("Remuxer is out of sync.");
+        if (Sage.DBG && tsSynced) System.out.println("Remuxer is out of sync.");
 
-        while (length - (offset + 377) > 0)
+        while (length - (offset + (188 * 2)) > 0)
```

Comment:
I'm fairly sure this needs to be 377, not 376 for the value. Because if length is 377 here and offset is zero, then the conditional is true. Then offset gets incremented to 1. Then it will index array element 377 (which I'm assuming could correspond to the length of that array in certain cases).